### PR TITLE
HTML Context

### PR DESCRIPTION
**Pull request by themadcreator**
Now typesettable supports SVG, Canvas, and HTML

### CODE REVIEW

File: src/contexts/html.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+ * license at https://github.com/palantir/typesettable/blob/develop/LICENSE
+ */
+
+import { IDimensions } from "../measurers";
+import { IAnchor, ITransform } from "../writers";
+import { ITypesetterContext } from "./index";
+
+export class HtmlUtils {
+  /**
+   * Appends an SVG element with the specified tag name to the provided element.
```
Comment (hellochar) [inline src/contexts/html.ts]: docs wrong: is not SVG

File: src/contexts/html.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+   * Returns the new element.
+   */
+  public static append(element: Element, tagName: string, ...classNames: string[]) {
+    const child = HtmlUtils.create(tagName, ...classNames);
+    element.appendChild(child);
+    return child;
+  }
+
+  /**
+   * Creates and returns a new SVGElement with the attached classnames.
```
Comment (hellochar) [inline src/contexts/html.ts]: docs here

File: src/contexts/html.ts [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+   */
+  public static create(tagName: string, ...classNames: string[]) {
+    const element = document.createElement(tagName) as HTMLElement;
```
Comment (hellochar) [inline src/contexts/html.ts]: I wonder if there's some way to proxy all the recognized built-in document.createElement() string literal types to the `tagName` so we don't have to cast

File: src/contexts/html.ts [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+      element.setAttribute("class", classNames.join(" "));
+    }
+  }
+
+  /**
+   * Returns the width/height of svg element's bounding box
+   */
+  public static getDimensions(element: HTMLElement): IDimensions {
+    // using feature detection, safely return the bounding box dimensions of the
+    // provided svg element
```
Comment (hellochar) [inline src/contexts/html.ts]: svg comment

File: src/contexts/html.ts [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+        element.classList.add(className);
+      });
+    } else {
+      // IE 11 does not support classList
+      element.setAttribute("class", classNames.join(" "));
+    }
+  }
+
+  /**
+   * Returns the width/height of svg element's bounding box
```
Comment (hellochar) [inline src/contexts/html.ts]: svg docs

File: src/contexts/html.ts [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+   */
+  public static getDimensions(element: HTMLElement): IDimensions {
+    // using feature detection, safely return the bounding box dimensions of the
+    // provided svg element
+    if (element.getBoundingClientRect) {
```
Comment (hellochar) [inline src/contexts/html.ts]: we shouldn't need feature detection on `getBoundingClientRect`; it's definitely supported in all the stuff we care about

File: src/contexts/html.ts [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+
+    // if can't get valid bbox, return 0,0
+    return { height: 0, width: 0 };
+  }
+}
+
+/**
+ * A typesetter context for HTML.
+ *
+ * @param element - The CSS font styles applied to `element` will determine the
```
Comment (hellochar) [inline src/contexts/html.ts]: What does `@param` on a class declaration mean?

File: src/contexts/html.ts (hunk 6 above)
Comment (themadcreator) [inline src/contexts/html.ts]: I've had some issues. I'm gonna be cautious

File: src/contexts/html.ts (hunk 7 above)
Comment (themadcreator) [inline src/contexts/html.ts]: woops

**Pull request merged**